dlrover/python/master/diagnosis/diagnosis_master.py

```diff
@@ -325,37 +313,9 @@ def start_observing(self):
             NodeInconsistencyDiagnostician(self._job_args),
             60 * 5,
         )
-
-        # TODO: need refactor
-        self._diagnostician.register_training_problems(
```

Comment:
Need to register the implement of 'hang detection' by using the new 'register_diagnostician'.

So u need to redefine a diagnostician for the current 'hang detection' by the following:
```
 Inference(
                    InferenceName.TRAINING,
                    InferenceAttribute.ISORNOT,
                    InferenceDescription.HANG,
                )
```